_rules/SC2-5-3-name-from-content-included-in-accessible-name.md

```diff
@@ -0,0 +1,176 @@
+
+## Assumptions
+
+This rule assumes that the [visible](#visible) [text](ttps://www.w3.org/TR/WCAG21/#dfn-text) has the same order as the text in the code, e.g. CSS has not been used to re-arrange pieces of text so that they are presented in a different order.
+
+## Accessibility Support
+
+There are no major accessibility support issues known for this rule.
+
+## Background
+
+- https://www.w3.org/TR/WCAG21/#label-in-name
+
+## Test cases
+
+### Passed
+
+#### Passed example 1
+
+Visible label and accessible name matches when trailing white spaces are removed.
+
+```html
+<div role="link" aria-label="next page ">next page</div>
+```
+
+#### Passed example 2
+
+Character insensitivity between visible label and accessible name.
+
+```html
+<div role="link" aria-label="Next Page">next page</div>
+```
+
+#### Passed example 3
+
+Full visible label is included in the accessible name.
+
+```html
+<button name="link" aria-label="Next Page in the list">Next Page</button>
+```
+
+#### Passed example 4
+
+Full visible label is included in the accessible name.
+
+```html
+<head>
+  <style>
+    .sr-only {
+      position:absolute;
+      left:-10000px;
+      top:auto;
+      width:1px;
+      height:1px;
+      overflow:hidden;}
+  </style>
+</head>
+
+<body>
+  <a href="home.html">Read more<span style="sr-only"> about our pricing</span></a>
+</body>
+```
+
+### Failed
+
+#### Failed example 1
+
+Visible label is not included in accessible name.
+
+```html
+<div role="link" aria-label="OK">Next</div>
+```
+
+#### Failed example 2
+
+Not all of visible label is included in accessible name.
+
+```html
+<button name="link" aria-label="the full">The full label</button>
+```
+
+#### Failed example 3
+
+The full visible label is split out across the accessible name.
+
+```html
+<head>
+  <style>
+    .sr-only {
+      position:absolute;
+      left:-10000px;
+      top:auto;
+      width:1px;
+      height:1px;
+      overflow:hidden;}
+  </style>
+</head>
+
+<body>
+  <a href="home.html">Link <span style="sr-only">that opens a new tab</span> to the homepage</a>
+</body>
+```
+
+### Inapplicable 
+
+#### Inapplicable example 1
+
+Not a widget role.
+
+```html
+<a aria-label="OK">Next</a>
+```
+
+#### Inapplicable example 2
+
+Widget role that does not support name from content.
+
+```html
+<input type="email" aria-label="E-mail" value='Contact'>
+```
+
+#### Inapplicable example 3
+
+Non-widget role that supports name from content.
+
+```html
+<div role="tooltip" aria-label="OK">Next</div>
+```
+
+#### Inapplicable example 4
+
+No rendered text in name from content.
+
+```html
+<div role="tooltip" aria-label="OK"></div>
+```
+
+#### Inapplicable example 5
+
+Sequence of characters in name from content is not expressing anything in human language and is therefore not considered [text](https://www.w3.org/TR/WCAG21/#dfn-text).
```

Comment:
I think this is debatable. X, in this context (and I would argue that human language interpretation is context dependent), expresses something.

Reply:
I think the argument here is that "X" is only used because it looks like a cross, so it actually means "cross/close", not "X" as the character.

@WilcoFiers and @ShadowBB, can you guys chime in on this discussion?

Reply:
There's definitely a point to be made here. In hindsight, I don't think this can be inapplicable. Applicability has to be objective, and determining whether or not "X" (or things like it) are meant as non-text content or as human language can be subjective. We should adjust this so it passes the rule.

I think that relates to my comment here:
https://github.com/auto-wcag/auto-wcag/pull/452/files#r269770000

Reply:
I agree with Wilco that this is hard to make objective in the applicability. So it should be moved to the expectations so we can subjectively pass it.

Just to be clear: x expresses something. But it does not express something in human language according to the definition in WCAG. It says so in the intent of this SC:

> For the purposes of this SC, text should not be considered a visible label if it is used in a symbolic manner, rather than directly "expressing something in human language" as per the definition of text in WCAG. For example, 1.4.5 Images of Text describes considerations for "symbolic text characters." In the images of text example "B", "I", and "ABC" appear on icons in a text editor, where they are meant to symbolize the functions for Bold, Italics, and Spelling, respectively. In such a case, the accessible name should be the function the button serves (e.g., "Spell check" or "Check spelling"), not the visible symbolic characters. A similar text editor is shown in Figure 1.
Icons for transforming text, including heading, bold, italic, quote, code, and link, along with icons for ordered and unordered lists and other controls
Likewise, where an author has used a greater-than symbol (">") to mimic the appearance of the right-facing arrow, the text does not convey something in human language. It is a symbol, in this scenario likely meant to mimic the icons used for a "Play" button or a "Next" arrow.

Reply:
I have tried moving the subjectiveness to the expectation now.